packages/core/contracts/oracle/implementation/VotingV2.sol

```diff
@@ -808,9 +808,15 @@ contract VotingV2 is Staker, OracleInterface, OracleAncillaryInterface, OracleGo
         // Traverse all requests from the last considered request. For each request see if the voter voted correctly or
         // not. Based on the outcome, attribute the associated slash to the voter.
         int256 slash = voterStake.unappliedSlash; // Load in any unapplied slashing from the previous iteration.
-        uint64 nextIndexToProcess = voterStake.nextIndexToProcess;
+
+        // If the voter has not yet staked, we must get the starting index to avoid utilising the default 0 index, which
+        // would lead to unitialized price requests being processed in the following loop.
+        uint64 nextIndexToProcess =
+            voterStake.nextIndexToProcess == 0 && voterStake.stake == 0
```

Comment:
Do we need to check that voterStake.stake == 0? If nextIndexToProcess is 0, we know we need to compute the initial index, no?

Reply:
Thanks @mrice32, good observation, 
`voterStake.stake == 0` seems redundant here but it's needed ONLY if VotingV2 is deployed with `startingIndex = 0` (as in all our tests). If `VotingV2.startingIndex=0`, then the first price request would have index 0 and a staked user `nextIndexToProcess` to process could be 0. In this scenario, if we don't check that `voterStake.stake == 0`, this initial users could skip the first slashing round. This can be verified by removing the `voterStake.stake` check and running the tests, one will fail.

If we decided to force the VotingV2 contract to always be deployed with `startingIndex !=0`, we could remove that extra check, but this would be a bigger change that would force us to update quite a few tests. 

What do you think? is it worth it? Gas-wise I don't think it would be a problem because of the order in which we check this: `voterStake.nextIndexToProcess == 0 && voterStake.stake`

cc. @chrismaree

Reply:
yup, @md0x you are totally correct that we only use the `stake` check when `startingIndex` is set to 0. I think we should support this as it is required on all of our tests and it seems like something we should support for easy of reasoning through the contracts. I think you should just add a small comment explaining why you need both checks and then we are good to go.

Reply:
I think this fix makes a lot of sense. Should we remove this line: https://github.com/UMAprotocol/protocol/blob/1433376751715769a53268f57daa157163dce800/packages/core/contracts/oracle/implementation/Staker.sol#L138. I think this line was meant to protect against the same thing, but this bug can be triggered by going through different entry points (like commit or calling updateTrackers directly).

Reply:
I think this line is intended to limit the number of price requests that a new user must process when initially joining and participating in the system by eliminating past price requests in which he was not staked. If we remove it, I think there may be cases where users have to process price requests in which they were not staked.